### PR TITLE
macOS aarch64 libsass for Apple M1

### LINKED ISSUE #48: sbt-sassify broken on Apple Silicon M1 + Big Sur
I'm seeing a JNA library loading issue on my MacBookAir with Apple Silicon (M1) + Big Sur.

Here is my full setup:
- Big Sur 11.2.1
- sbt 1.4.7
- scala 2.13.5
- PlayFramework 2.8.7
- OpenJDK 11.0.9+11 (`/opt/homebrew/Cellar/openjdk@11/11.0.9/libexec/openjdk.jdk/Contents/Home`)
- sbt-sassify 1.5.1

Here's the full stack trace when running `last sassify`
```
[info] Sass compiling on 24 source(s)
[error] java.lang.UnsatisfiedLinkError: /Users/benjamin/Library/Caches/JNA/temp/jna17537362502519181591.tmp: dlopen(/Users/benjamin/Library/Caches/JNA/temp/jna17537362502519181591.tmp, 1): no suitable image found.  Did find:
[error] 	/Users/benjamin/Library/Caches/JNA/temp/jna17537362502519181591.tmp: no matching architecture in universal wrapper
[error] 	/Users/benjamin/Library/Caches/JNA/temp/jna17537362502519181591.tmp: no matching architecture in universal wrapper
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
[error] 	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
[error] 	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
[error] 	at java.base/java.lang.Runtime.load0(Runtime.java:768)
[error] 	at java.base/java.lang.System.load(System.java:1837)
[error] 	at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:1018)
[error] 	at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:988)
[error] 	at com.sun.jna.Native.<clinit>(Native.java:195)
[error] 	at org.irundaia.sass.SassCompiler$.<init>(SassCompiler.scala:28)
[error] 	at org.irundaia.sass.SassCompiler$.<clinit>(SassCompiler.scala)
[error] 	at org.irundaia.sbt.sass.SbtSassify$.$anonfun$baseSbtSassifySettings$10(SbtSassify.scala:100)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:285)
[error] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:285)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:278)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at org.irundaia.sbt.sass.SbtSassify$.$anonfun$baseSbtSassifySettings$8(SbtSassify.scala:100)
[error] 	at com.typesafe.sbt.web.incremental.package$.syncIncremental(package.scala:228)
[error] 	at org.irundaia.sbt.sass.SbtSassify$.$anonfun$baseSbtSassifySettings$6(SbtSassify.scala:140)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:834)
```

Note: I'm seeing a [similar issue](https://discuss.lightbend.com/t/apple-silicon-m1-playframework-broken-on-apple-silicon/7924) with the [directory-watcher](https://github.com/gmethvin/directory-watcher) used by [play-file-watcher](https://github.com/playframework/play-file-watch).

Other related issues:
* [directory-watcher unable to load Carbon library via JNA](https://github.com/gmethvin/directory-watcher/issues/52)
* [Big Sur changes to how JNA loads dylibs](https://github.com/java-native-access/jna/pull/1216)
* [PlayFramework runner failing due to directory-watcher/Carbon/JNA issue](https://github.com/playframework/playframework/issues/10372)
* [Discussion on HackerNews about Mac OS 11 loading dynamic libraries](https://news.ycombinator.com/item?id=23612772)

### PR DESCRIPTION
**Pull request by porchy13**
This PR should close #48.

To get `sbt-sassify` working on Apple M1 computers and using `sbt 1.6.0-M1` needs to have `libsass.dylib` in arch specific folders: `darwin-aarch64` for M1 chips and `darwin-x86-64` for Intel chips. The old `darwin` folder needs to be kept for `sbt` version prior to 1.6.

I did not find any ways to have `libsass` cross-compiled for `aarch64`. The one I provide comes from my Homebrew installation.

Finally, I have update some dependencies and the test protocol has been fixed in that sense.

### CODE REVIEW

**Comment by thadamski** [comment]
Thanks for creating a PR! Is there a beta build of this branch available in a repo? I'd like to help test, as I'm also experiencing the problems referenced in https://github.com/irundaia/sbt-sassify/issues/48 as well

**Comment by porchy13** [comment]
I've not publish this version in a public repository. I do the following steps to use it in my projects:
1. forked the project and publish the plugin locally (`sbt clean; compile; publishLocal`).
2. copy the local `sbt-sassify.jar` (found in `~/.ivy2/local/org.irundaia.sbt/sbt-sassify/scala_2.12/sbt_1.0/1.5.2-SNAPSHOT/jars`) in `project/lib`.
3. in `project/plugins.sbt`, I use the local plugin by declaring it like below:
   ```sbt
   val sassifyLocation: File = file("project/lib/sbt-sassify.jar").getAbsoluteFile
   addSbtPlugin("org.irundaia.sbt" % "sbt-sassify" % "1.5.2-SNAPSHOT" from "file://" + sassifyLocation)
   ```

**Comment by jwaring** [comment]
Thanks @porchy13 for creating the PR. I have followed the solution as you documented above and it works well.
I look forward to the PR being incorporated into the next release.

**Comment by arturopala** [comment]
@irundaia any chance to get this fix released?

File: project/build.properties [hunk 1]
```diff
@@ -1 +1 @@
-sbt.version=1.3.10
+sbt.version=1.6.1
```
Comment (mkurz) [inline project/build.properties]: 1.6.2 released

Comment (porchy13) [reply project/build.properties]: 👍 Updated in my last commit.

**Comment by irundaia** [comment]
Sorry, I've been away for a while. Question, how did you compile the darwin-arm version of libsass?

**Comment by porchy13** [comment]
I've used the binaries provided by [Homebrew (link to the libsass formulae)](https://formulae.brew.sh/formula/libsass).

I did not find any cross-compilation Docker image that was able to compile for the `aarch64` architecture (all the tests I made have failed).

**Comment by irundaia** [comment]
I've tested in on my x86 Mac. It still works there. So I'll try to release this this evening, after doing some test on a windows machine.

**Comment by arturopala** [comment]
@irundaia hello, do you plan to release a new version of the library on maven central?

**Comment by irundaia** [comment]
@arturopala, I do. I’m in the process of figuring out how to release the plugin to jCenter now that bintray has shut down.

**Comment by arturopala** [comment]
Would it be an option to deploy to the Sonatype OSS? If so, I have an
example in
https://github.com/arturopala/buffer-and-slice/actions/workflows/publish.yml

śr., 2 mar 2022 o 13:31 Han van Venrooij ***@***.***>
napisał(a):

**Comment by irundaia** [comment]
I've published the version 1.5.2 to Sonatype OSS. 

Due to Sonatype's stricter groupId policies, I had to change the groupId to `io.github.irundaia`. So to upgrade, change your `project/plugins.sbt` to `addSbtPlugin("io.github.irundaia" % "sbt-sassify" % "1.5.2")`

**Comment by arturopala** [comment]
Thank you @irundaia !

**Comment by jwaring** [comment]
My thanks also @irundaia and @porchy13 for the original PR

**Comment by mkurz** [comment]
@irundaia You should also let scala-steward know about the new groupid: https://github.com/scala-steward-org/scala-steward/pull/2488

**Pull request merged**